### ISSUE TITLE
Need help using vouch with EKS, AWS cognito and ALB ingress

### ISSUE THREAD

**Issue by vumdao** [vouch/vouch-proxy#483]
I got 302 redirect when accessing app domain

![image](https://user-images.githubusercontent.com/37215642/177705252-2bf4e2ac-2069-44ea-9fa1-f52efe5ec907.png)

Logs, nginx config, vouch config and services [here](https://gist.github.com/vumdao/92405b282c05182d29be0c191fa2552e#file-eks-cognito-vouch-md)

**Comment by bnfinet** [comment]
That's the correct behavior in testing mode.  You should click on the last link to continue.

**Comment by vumdao** [comment]
@bnfinet I click the last link which is the login page of cognito. After login, it redirect to callback URL which is `https://dev-vouch.cloudopz.co/auth` but not yet go to my app page. What should I do next?

Thanks.

**Comment by bnfinet** [comment]
That's another 302, yes?  Click the last link again to go to your app.

**Comment by vumdao** [comment]
@bnfinet 
it's another 302 and if I click the last link again, it just double the line
![image](https://user-images.githubusercontent.com/37215642/177764952-4936df1a-6047-4719-b2aa-898eacb65e0a.png)

**Comment by bnfinet** [comment]
Please update your logs and config in the gist.  Please provide complete logs from startup through all the click throughs.  Do not put them in an md file.  Please make sure that line numbers are displayed.

**Comment by bnfinet** [comment]
Use 'cloudopz.co' as your domain.
https://gist.github.com/vumdao/0a788b4ff24279dc03462028ae68a10e#file-config-yml-L6

Remove 'cookie.secure: false' since you're using https.
https://gist.github.com/vumdao/0a788b4ff24279dc03462028ae68a10e#file-config-yml-L12

Remove any config item which is set to a default value.

Keep your config as simple as possible.

Those logs don't show the redirects through /auth.  Please click through those links.  Are you including all logs?  If you're getting redirected back to /login please include at least two complete round trips with all logs.

**Comment by vumdao** [comment]
@bnfinet I passed the validate and can go to the app URL, but the app domain just redirect to `dev-solr.cloudopz.co:8080/solr`
Is there any wrong in my app location at nginx.conf ?

New [vouch.log](https://gist.github.com/vumdao/63cd3c636705bb421b7d881ad1f5b4b6)

[nginx.conf](https://gist.github.com/vumdao/44cee1dcffee4d0af7485caeca63a115)

Thanks

**Comment by bnfinet** [comment]
@vumdao it's really hard to follow your setup.  It appears that your domain names are different now and your VP config has changed.  That nginx config does not correspond to the logs you're offering.

> One more question: Why do we need to add email domain tho the vouch whitelist domain? Since the email is register with AWS cognito with any domain

VP is meant to protect one or more domains.  The VP cookie is set into one domain.  The cookie domain, the domain that VP is served from and the protected app must all be the same domain, such as cloudopz.co

**Comment by vumdao** [comment]
@bnfinet My domain is not changed, it is still `cloudopz.co` as you see in the log.

The log and the nginx.conf was what I used to test it, why do you see they are not related to each other? Please help to get the latest one, I had to change nginx.conf to fixed the validate flow

New [vouch.log](https://gist.github.com/vumdao/63cd3c636705bb421b7d881ad1f5b4b6)
[nginx.conf](https://gist.github.com/vumdao/44cee1dcffee4d0af7485caeca63a115)

**Comment by bnfinet** [comment]
> The log and the nginx.conf was what I used to test it, why do you see they are not related to each other? Please help to get the latest one, I had to change nginx.conf to fixed the validate flow

In your prior gist you showed two different domain..

This line shows the domain `vincere.io`
https://gist.github.com/vumdao/44cee1dcffee4d0af7485caeca63a115#file-nginx-conf-L30

But the log shows `cloudopz.co`
https://gist.github.com/vumdao/63cd3c636705bb421b7d881ad1f5b4b6#file-vouch-log-L21-L22

The new logs and nginx config show VP working correctly and your app passing validation.

Is it working for you??  Looks like you've changed from https to http, is that right? Did you change your VP config?   Please add your current VP config to a gist.

FYI - I'm going to be stepping away from the computer here and won't be able to offer you any support today.

**Comment by vumdao** [comment]
@bnfinet oops! sorry I forgot to obfuscate the nginx.conf, just tried to hide the real domain.

Yep, I change from http to https, maybe I will add back the `cookie.secure: false`, but for now as you see the validation works well, it just got stuck at https://gist.github.com/vumdao/44cee1dcffee4d0af7485caeca63a115#file-nginx-conf-L37 

After login, I open `https://dev-solr.cloudopz.co` but it redirects to `http://dev-solr.cloudopz:8080/solr/`

**Comment by vumdao** [comment]
@bnfinet the issue was fixed after I add more config to app location 
```
   location / {
      proxy_pass http://solr:8983;
      proxy_set_header X-Vouch-User $auth_resp_x_vouch_user;
      proxy_set_header Host $http_host;
      proxy_pass_request_body off;
      proxy_set_header Content-Length "";
      auth_request_set $auth_resp_x_vouch_user $upstream_http_x_vouch_user;
      auth_request_set $auth_resp_jwt $upstream_http_x_vouch_jwt;
      auth_request_set $auth_resp_err $upstream_http_x_vouch_err;
      auth_request_set $auth_resp_failcount $upstream_http_x_vouch_failcount;
   }
```

**Comment by bnfinet** [comment]
@vumdao I'm glad that you did get it working

By not passing the request body to your app HTTP POST requests will usually not work.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST

Those last three items in that stanza are extraneous.  In general, I would suggest that you try to remove as much config as possible.

Best of luck!
